X509_up_ref has been available since LibreSSL 2.5.
X509_free has always been available.

Index: content/fetchers/curl.c
--- content/fetchers/curl.c.orig
+++ content/fetchers/curl.c
@@ -96,9 +96,8 @@
 #include <openssl/x509v3.h>
 
 /* OpenSSL 1.0.x to 1.1.0 certificate reference counting changed
- * LibreSSL declares its OpenSSL version as 2.1 but only supports the old way
  */
-#if (defined(LIBRESSL_VERSION_NUMBER) || (OPENSSL_VERSION_NUMBER < 0x1010000fL))
+#if (OPENSSL_VERSION_NUMBER < 0x1010000fL)
 static int ns_X509_up_ref(X509 *cert)
 {
 	cert->references++;
